### PR TITLE
add support for k8s 1.25.0

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

TODO: needs an update in the control-plane tests. Will be done as a follow up

### CODE REVIEW

**Comment by nbusseneau** [comment]
/test-1.25-net-next

**Comment by nbusseneau** [comment]
/test-1.25-net-next

**Comment by nbusseneau** [comment]
/test-1.25-net-next

**Comment by nbusseneau** [comment]
/test-1.25-net-next

**Comment by nbusseneau** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.25-net-next

**Comment by aanm** [comment]
/test-1.24-net-next

File: test/k8s/manifests/1.21/external_endpoint.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: discovery.k8s.io/v1beta1
```
Comment (sayboras) [inline test/k8s/manifests/1.21/external_endpoint.yaml:1]: AFAIK EndpointSlice becomes stable in 1.21, any reason we are still using v1beta1 here?

https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/

**Review by sayboras — APPROVED** [review]
LGTM 💯 

For other reviewer, IngressClass is removed from slim objects, as we are creating Cilium IngressClass as part of helm instead.

**Review by tklauser — APPROVED** [review]

**Review by rolinh — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]
Yes.

**Pull request merged**